src/Compilers/CSharp/Test/Semantic/Semantics/NullableReferenceTypesTests.cs

```diff
@@ -53389,11 +53585,14 @@ void M1(object? x1, object? y1)
     ref U M2<U>(U a) where U : object => throw null;
 }
 ";
-
+            // Note: you cannot pass a `T` to a `U : object` even if the `T` was null-tested
             CreateCompilation(source, options: WithNonNullTypesTrue()).VerifyDiagnostics(
-                // (8,18): warning CS8601: Possible null reference assignment.
```

Comment:
>// (8,18): warning CS8601: Possible null reference assignment. [](start = 16, length = 62)

When a warning like this disappears because we no longer infer not nullable type, I think we should change the scenario to an attempt to dereference the result, so that we can observe the result of inference.